### DISCUSSION TITLE
Allow mappers to change difficulty values throughout a map

### DISCUSSION THREAD

**Discussion by Doksha** [ppy/osu#20020] [Ideas] [answered]
Give mappers the ability to dynamically alter the difficulty throughout a map (namely CS and OD) 
Maps with this option would be unrankable, but would be cool as loved maps. This could allow for a single marathon to cover many skillsets without needing to be split between difficulties for precision and accuracy based skillsets. In general, this would allow for a ton of creativity in mapping, and allow for tons of fun concepts and new patterns.

This could be implemented by storing these values in a separate section in the map file, similar to timing points, making it so these maps would still be compatible with stable (just not with the custom values) The 'classic' mod would also ignore these custom values

Star rating would be calculated ignoring custom values, since the SR system would likely need an entire rework to account for dynamic difficulty.

This may have been requested before but I don't know what to search to find it.

**Comment by peppy** [comment, marked as answer]
Internally we've already got most things in place to make this happen. There's also plenty of existing issues discussing this if you find the correct keywords.

It will be one of those things we get to adding once we have a baseline editor workflow figured out (including a new serialisation format).
